maven-core/src/main/java/org/apache/maven/graph/ProjectsCollector.java

```diff
@@ -0,0 +1,43 @@
+package org.apache.maven.graph;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.apache.maven.execution.MavenExecutionRequest;
+import org.apache.maven.project.MavenProject;
+import org.apache.maven.project.ProjectBuildingException;
+
+import java.io.File;
+import java.util.List;
+
+/**
+ * Facade to collect projects for a given set of pom.xml files.
+ */
+public interface ProjectsCollector
+{
+    /**
+     * Collect Maven projects from a list of POM files.
+     * @param projects List that will be filled with the found projects.
+     * @param files List of POM files.
+     * @param request The {@link MavenExecutionRequest}
+     * @throws ProjectBuildingException In case the POMs are not used.
+     */
+    void collectProjects( List<MavenProject> projects, List<File> files, MavenExecutionRequest request )
+            throws ProjectBuildingException;
```

Comment:
Shouldn't this be a an exception in the same package (or should ProjectsCollector be moved to `org.apache.maven.project`?

Reply:
Same as above: I've moved the Collector related classes to a new package within the project package.